### PR TITLE
Bump upjet to commit bd528e443b6f and bump Terraform CLI to v1.5.5

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

Bumps upjet to commit `bd528e443b6f`. 
Bumps the Terraform CLI to `v1.5.5` and consumes the Terraform CLI from https://github.com/upbound/terraform/releases.

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

Via uptest

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

File: cluster/images/provider-azuread/Dockerfile
Comment (ulucinar) [inline cluster/images/provider-azuread/Dockerfile:1]: Could you please bump the `alpine` version to `v3.18.3` to consume the available security fixes?

**Comment by ulucinar** [comment]
/test-examples="examples/applications/application.yaml"

**Review by ulucinar — APPROVED** [review]
Thanks @turkenf, lgtm.

**Pull request merged**